src/lock.rs

```diff
@@ -153,49 +118,94 @@ impl Lock {
                 name: name.to_string(),
             });
         }
-        Ok((changed, res))
+        Ok(changed)
+    }
+
+    pub fn freeze(&mut self, deps: &[FrozenDependency]) {
+        self.patch_existing_deps(&deps);
+        self.add_missing_deps(&deps);
+    }
+
+    fn add_missing_deps(&mut self, frozen_deps: &[FrozenDependency]) {
+        let known_names: &Vec<_> = &mut self.dependencies.iter().map(|d| d.name()).collect();
+        let new_deps: Vec<_> = frozen_deps
+            .iter()
+            .filter(|x| !known_names.contains(&&x.name))
```

Comment:
... you're just doing it on purpose, admit it =] ?
x is already a double-ref due to the iter(), so you're taking a `&&str` from a `&&FrozenDependency`. You can remove one `&`.

Reply:
Hmm, in case you're replying to the other comment on line 166, I'm mostly concerned that you don't check the return value at all. If it's None you ignore it.

So then you might as well really return nothing, if you're not going to handle the result, yes?